### PR TITLE
Header/responsive styling

### PR DESCRIPTION
**Pull request by tammalee**
## What was done
- header styling focus outline now green 
- button is primary with white for better contrast 
- button goes to ghost on hover, kas is deciding on this 
- back to jobber link has border bottom on hover, kas is deciding on this 
- mobile pop-under has a border top and box shadow along the bottom 
- values converted to variables where it  makes sense for header 
- responsive down to 320px width

## What wasn't done
- final decisions on hover states for button and back to Jobber link.

## What to test
Please test the line items under What was done. If there are changes to make to buttons/links after Kas has had time to think about it, there will be a new PR to address those changes.

### CODE REVIEW

**Comment by eddysims** [comment]
not sure if the alignment is meant to be finished, but its quite off.

image

image

**Comment by eddysims** [comment]
There is no transition on the chat now background, only the text color

**Comment by eddysims** [comment]
Axe warning on the avatar

image

File: styles/_base.scss [hunk 1]
```diff
@@ -48,6 +48,9 @@ a {
   &:focus {
     text-decoration: underline;
   }
+  &:focus {
```
Comment (eddysims) [inline styles/_base.scss]: missing a space here

File: styles/_header.scss [hunk 2]
```diff
@@ -1,42 +1,43 @@
 /***** Header *****/
-
-
 .header {
   background-color: $white;
   margin: 0 auto;
   max-width: $max-width-page-container;
+  box-shadow: 0 3px 5px hsla(0,0%,72.2%,.15);
```
Comment (eddysims) [inline styles/_header.scss]: one zero is unitless, one is %.

0 should always be unitless.

File: styles/_header.scss [hunk 3]
```diff
@@ -1,42 +1,43 @@
...
-      border-right: solid 2px $primary;
-      margin-right: 16px;
+  &__link {
+    &:hover {
+      text-decoration: none;
     }
+  }
+
+  &__image {
+    @include fluid-value( max-height, $min-width, $max-width, 18.50px, 35px);
```
Comment (eddysims) [inline styles/_header.scss]: `18.50px` should be rounded. There is no such thing as a real half pixel

File: styles/_header.scss [hunk 4]
```diff
@@ -49,15 +50,15 @@
...
   &[aria-expanded="true"] {
-    background-color: #fff;
-    box-shadow: 0 0 4px 0 rgba(0, 0, 0, .15), 0 4px 10px 0 rgba(0, 0, 0, .1);
-    border: solid 1px #ddd;
     right: 0;
     left: 0;
-    @include fluid-value( top, $min-width, $max-width, 125px, 138px);
+    @include fluid-value( top, $min-width, $max-width, $header__height--min, $header__height--max);
     z-index: 1;
+    background-color: #fff;
```
Comment (eddysims) [inline styles/_header.scss]: color should be a variable

File: styles/_header.scss [hunk 5]
```diff
@@ -49,15 +50,15 @@
...
+    border-top: solid $border--width $primary;
+    box-shadow: 0 3px 5px hsla(0,0%,72.2%,.15);
```
Comment (eddysims) [inline styles/_header.scss]: zero should be unitless

File: styles/_header.scss [hunk 6]
```diff
@@ -68,52 +69,65 @@
 
 .nav-wrapper {
   a {
```
Comment (eddysims) [inline styles/_header.scss]: can we target a class and not the tag?

File: styles/_header.scss [hunk 7]
```diff
@@ -68,52 +69,65 @@
...
+      &:focus,
+      &:active {
+        background-color: $white;
+        color: $primary;
+        text-decoration: none;
+        border: solid $border--width $primary;
+      }
     }
 
+    [dir="rtl"] & { padding: 0 0 0 20px;}
```
Comment (eddysims) [inline styles/_header.scss]: missing space before `}`

File: styles/_variables.scss [hunk 8]
```diff
@@ -85,4 +85,11 @@ $spacing--desktop: 30px;
 
 $border--width: 1px;
 $border--radius: 8px;
+$focus--width: 5px;
 $font-zendesk-icons: "copenhagen-icons";
+
+/* HEADER SPECIFIC VARIABLES */
+$header__height--min: 80px;
```
Comment (eddysims) [inline styles/_variables.scss]: I recall you saying we were going to us `em`s why are these all `px` based?

Comment (tammalee) [reply styles/_variables.scss]: Ems don't work with the mixin I'm using.
I could probably refactor that.

**Comment by tammalee** [comment]
> not sure if the alignment is meant to be finished, but its quite off.
> 
> image
> 
> image

What is the viewport size? It should be good down to 320px.

File: styles/_header.scss (hunk 6 above)
Comment (tammalee) [reply styles/_header.scss]: Sure. There will be parts of this codebase that I likely won't refactor to do that. Just warning you!

File: styles/_header.scss (hunk 3 above)
Comment (tammalee) [reply styles/_header.scss]: _facepalm_

**Pull request merged**